basics/src/data.rs

```diff
@@ -1063,19 +1063,19 @@ mod tests {
 
         let rec = vec![DataType::Int(5), "asdfasdfasdfasdf".into(), "asdf".into()];
 
-        // DataType should always use 16 bytes itself
-        assert_eq!(size_of::<DataType>(), 16);
-        assert_eq!(size_of_val(&txt), 16);
+        // DataType should always use 24 bytes itself
+        assert_eq!(size_of::<DataType>(), 24);
```

Comment:
Ouch... This is pretty bad :/ Any way we can avoid it?

Reply:
I think the way to avoid it is to make `AutoIncrementRequest` a `u64` rather than a `(u32, u64)`, and to store the shard ID in the high bits of the `u64`. If we assume < 256 shards, we can do 48 bits of ID and 8 bits of shard ID.

Reply:
Introducing the first `DataType::ID` type didn't increase the overall size of `DataType` — `DataType::AutoIncrementRequest` did, which is just an empty variant. I wonder if it might be because it takes `DataType` above 8 enum variants?

Could perhaps make an `AutoIncrement` enum with an `ID` and a `Request`  variant, but not sure how nice that is either.

Reply:
The enum should support 256 variants before needing a larger variant tag. The issue is actually due to alignment rather than overall data size. AutoIncrementID is 8 byte aligned, so it includes an additional 4 bytes of padding, beyond the 12 bytes of data. Counting the variant tag, it spills over beyond 16 bytes so the enum spills over into taking 24 bytes.

Reply:
I think the best solution would just be to change the ID variant to `ID(u32, u64)` and then do the conversion to AutoIncrementID whenever removing the value from the enum

Reply:
Thanks @fintelia — that makes sense. Moved back to what you're suggesting in the last commit.